iceoryx_hoofs/source/posix_wrapper/named_pipe.cpp

```diff
@@ -153,9 +155,18 @@ NamedPipe::~NamedPipe() noexcept
 template <typename Prefix>
 IpcChannelName_t NamedPipe::convertName(const Prefix& p, const IpcChannelName_t& name) noexcept
 {
-    return IpcChannelName_t(cxx::TruncateToCapacity,
-                            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
-                            cxx::concatenate(p, (name.c_str()[0] == '/') ? *name.substr(1) : name).c_str());
+    IpcChannelName_t channelName = p;
+
+    if (name[0] == '/')
+    {
+        name.substr(1).and_then([&](auto& s) { channelName.append(cxx::TruncateToCapacity, s); });
```

Comment:
Shouldn't we at least drop a warning or information to the user that we remove the leading slash here?

Reply:
No not at all. In the end the ipc channel contract is to provide a valid file name (which does not allow slashes) in the IPC channel.
This still needs to be refactored somehow, since some IPC channels require slashes (message queue), some allow full paths (named pipe) and some only require a valid file name (unix domain socket).

This all has to be handled by an abstraction above those constructs which will be introduced in: https://github.com/eclipse-iceoryx/iceoryx/pull/1451

Until then I would leave it as it is and then refactor it step by step especially since the `NamedPipe` is only used in Windows.